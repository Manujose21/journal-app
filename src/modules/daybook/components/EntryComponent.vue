<template>
  <div 
    class="mb-3 pointer" 
    @click="$router.push({ name: 'entry-view', params:{ entry: entry.id } })"
  >

    <div class="entry-container p-2">

      <div class="entry-title">
        <span class="fw-bold mx-1">{{ date }}</span>
      </div>

      <div class="entry-description mt-1">
        <p>{{ entryText }}</p>
      </div>

    </div>
  </div>
</template>

<script>
  import getDayMonthYear from '../helpers/getDayMonthYear.js'
 
  export default {
    props:{
        entry:{
          type : Object,
          reuired: true
        }
    },

    computed:{
      entryText(){
        const text = this.entry.text
        if( text.length > 130 ) return text.substring(0,130)+"..."
        return text
      },
      date(){
        return getDayMonthYear(this.entry)
      }
    }
    
  }
</script>

<style lang="scss" scoped>
  .entry-container{
      border-bottom: 1px solid #ccc;
      transition: background-color 0.3s;
      &:hover{
        background-color: lighten( $color:grey, $amount: 45%);
      }
      .entry-description{
        font-size: 12px;
      }
  }
</style>