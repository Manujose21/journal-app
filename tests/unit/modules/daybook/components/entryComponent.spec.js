import { shallowMount } from '@vue/test-utils'
import EntryComponent from '@/modules/daybook/components/EntryComponent.vue'
import { daybookState } from '../../../mock-data/test-data'

describe('tests entry component', ()=>{
  
  const mockRouter = {
    push: jest.fn()
  }
  
  const wrapper = shallowMount(EntryComponent, {
    props:{
      entry : daybookState.entries[0]
    },
    global:{
      mocks:{
        $router: mockRouter
      }
    }
  })
  
  
  it(' match with the snapshot ', () => {
  
    expect(wrapper.html()).toMatchSnapshot()

  })

  it('should redirect', () => {

    wrapper.find('.pointer').trigger('click')

    expect(mockRouter.push).toHaveBeenCalledWith({
      name: 'entry-view',
      params:{
        entry: daybookState.entries[0].id
      }
    })

  })

  it('date compuputed', () => {

    expect(wrapper.vm.date).toBe('23 Septiembre 2022, Viernes ')

  })

})
