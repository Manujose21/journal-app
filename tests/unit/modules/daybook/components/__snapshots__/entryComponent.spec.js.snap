// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests entry component  match with the snapshot  1`] = `
<div class="mb-3 pointer">
  <div class="entry-container p-2">
    <div class="entry-title"><span class="fw-bold mx-1">23 Septiembre 2022, Viernes </span></div>
    <div class="entry-description mt-1">
      <p>estoy aprendiendo vue y vuex</p>
    </div>
  </div>
</div>
`;
